### ISSUE TITLE
PointsToGraph goes out of memory

### ISSUE THREAD

**Issue by zhaogang92** [secure-software-engineering/phasar#23]
I wrote a simple dataflow analysis with the IFDS framework. Then I tried it with the gcc benchmark. The points-to grpah construction just went out of memory (48GB). Is this normal? Is any quick workaround?

**Comment by pdschubert** [comment]
We already observed the same for some programs which involve a massive amount of pointers. When we tried to analyze the clang compiler which involves around 5 million pointers (in LLVM IR), we noticed that an analysis exceeds 128 GB of RAM. Currently, we are looking to solve this problem by only constructing the points-to information that is actually needed in an on-the-fly manner.

**Comment by zhaogang92** [comment]
The major bottleneck is the boost graph structure in my case. Yesterday I replaced the adjacency_list with this compressed_sparse_row_graph and it saves lots of memory (though it's problematic today under Release mode, I don't know where is messed up). I only changed a few classes so I am not sure if is a solution. And it's much slower than adjacency_list. 

I also wondered if the adjacency_list of boost 1.66 is buggy. Their memory consumption difference is too large (48GB v.s. less than 8GB). I even don't know why 0.7 million instructions need to take more than 48GB memory by that graph. So I tried to compile Phasar with different boost versions, but always errors.

BTW, in my case (gcc benchmark), the default CFLAndersAAWP used in ProjectIRDB will let the analysis get stuck in the AA.alias function when constructing the PointsToGraph. After analyzing several functions, that function will never return and consume all of the memory. I replaced it with CFLSteensAAWP then this problem is gone. Maybe it's just a special case.
